### PR TITLE
Adjustments Needed for Skillmodels + Bug Fixes

### PR DESCRIPTION
**Pull request by janosg**
## Bug

The new `process_constraints` sometimes fixes all equality constrained parameters, but the first one has to remain free. This is due to a bug in the consolidation of equality constraints. 

## Skillmodels Adjustment

In skillmodels I have to call `process_constraints` with a params DataFrame where all entries in the "value" columns are np.nan. This raised an error because the constraints are not fulfilled. Now the checks are only performed if at least one "value" is not missing.

### CODE REVIEW

File: estimagic/optimization/consolidate_constraints.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def _consolidate_fixes_with_equality_constraints(fixed_pc, equality_pc, params):
             assert (
                 len(valcounts) == 1
             ), "Equality constrained parameters cannot be fixed to different values."
-            fixed_value.iloc[eq["index"]] = valcounts.idex[0]
```
Comment (tobiasraabe) [inline estimagic/optimization/consolidate_constraints.py]: Ups. Sorry, I have overlooked this gem in the last PR, right?🤦‍♂

File: estimagic/optimization/check_constraints.py [hunk 2]
```diff
@@ -16,6 +16,9 @@ def check_constraints_are_satisfied(pc, params):
     This should be called before the more specialized constraints are rewritten to
     linear constraints in order to get better error messages!
 
+    We let the checks pass if all parameter values are np.nan. This is useful for the
+    construction of star parameters.
```
Comment (tobiasraabe) [inline estimagic/optimization/check_constraints.py]: What are star parameters?

File: estimagic/optimization/consolidate_constraints.py (hunk 1 above)
Comment (janosg) [reply estimagic/optimization/consolidate_constraints.py]: Yes ;-)

File: estimagic/optimization/check_constraints.py (hunk 2 above)
Comment (janosg) [reply estimagic/optimization/check_constraints.py]: They are star**t** parameters without a t in the end ...

**Pull request merged**